### DISCUSSION TITLE
Is it possible to change an environment variable based on user input?

### DISCUSSION THREAD

**Discussion by zmzlois** [vercel/community#1224] [Help] [answered]
### Summary

Hi team,

I am currently working on a project that will prompt users for a URL of their file and I will need to parse part of their URL and put them into an environment variable. At the same time I need it to be encrypted so I won't see it. Is it achievable? If yes, how? If no, will it be added to product roadmap?

I have gone through the environment variable file and don't think it is available.

Best,
Lois

**Comment by amyegan** [comment, marked as answer]
Hi @zmzlois! Environment Variable changes require redeployment, which doesn't sound like it would work for your site. But the new [Edge Config](https://vercel.com/docs/concepts/edge-network/edge-config#) or one of the [database integrations](https://vercel.com/integrations#databases) might work for you.
